### PR TITLE
[MO] - [Azure] -> MM pipeline paralelism disabled

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: morsak <xorsak02@stud.fit.vutbr.cz>

### Type of change

- Bugfix

### Description

Same problem as with https://github.com/strimzi/strimzi-kafka-operator/pull/4806 PR. Mirror maker regression pipeline sometimes end up because of the OOM problem. 

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

**Review by Frawless — APPROVED** [review]

File: .azure/regression-pipeline.yaml [hunk 1]
```diff
@@ -62,6 +62,7 @@ jobs:
       test_case: 'mirrormaker/**/*ST,kafka/dynamicconfiguration/**/*ST'
       groups: 'regression'
       cluster_operator_install_type: 'bundle'
+      run_parallel: false
```
Comment (Frawless) [inline .azure/regression-pipeline.yaml:65]: do we really need to set this? We use also `parallel` so maybe it makes sense to set it to `1` ?

Comment (see-quick) [reply .azure/regression-pipeline.yaml:65]: Yes, we do. Based on my observation I saw some edge cases when you allow paralelism `(junit.jupiter.execution.parallel.enabled=true)` and let `( junit.jupiter.execution.parallel.config.fixed.parallelism=1)` can run 2 tests at once. It is maybe bug inside `junit` but practically speaking it can be resolved but setting this flag to `false`. Moreover it ensures that only `main` thread will executed tests instead of `fork-join` pool thread selected by JVM.
